examples/mixture/plot_gmm_sin.py

```diff
@@ -3,15 +3,16 @@
 Gaussian Mixture Model Sine Curve
 =================================
 
-This example highlights the advantages of the Dirichlet Process:
-complexity control and dealing with sparse data. The dataset is formed
-by 100 points loosely spaced following a noisy sine curve. The fit by
-the GMM class, using the expectation-maximization algorithm to fit a
-mixture of 10 Gaussian components, finds too-small components and very
-little structure. The fits by the Dirichlet process, however, show
-that the model can either learn a global structure for the data (small
-alpha) or easily interpolate to finding relevant local structure
-(large alpha), never falling into the problems shown by the GMM class.
+This example highlights the advantages of the Dirichlet Process: complexity
+control and dealing with sparse data. The dataset is formed by 100 points
+loosely spaced following a noisy sine curve. The fit by the ``GaussianMixture``
+class, using the expectation-maximization algorithm to fit a mixture of 10
+Gaussian components, finds too-small components and very little structure. The
+fits obtained by the ``BayesianGaussianMixture`` class with a Dirichlet
+process prior, however, show that the model can either learn a global structure
+for the data (small ``weight_concentration_prior``) or easily interpolate to
+finding relevant local structure (large ``weight_concentration_prior``), never
+falling into the problems shown by the ``GaussianMixture`` class.
```

Comment:
I don't agree with this analysis. Let me suggest the following instead:

This example demonstrates the behavior of Gaussian mixture models on data that was not generated by a mixture of Gaussian random variables.  The dataset is formed by 100 points loosely spaced following a noisy sine curve.  There is therefore no ground truth value for the number of Gaussian components.

The first model is a classical Gaussian Mixture Model with 10 components fit with the Expectation Maximization algorithm.

The second model is a Bayesian Gaussian Mixture Model  with a Dirichlet process prior fit with variational inference. The low value of the concentration prior makes the model favor a lower number of active components. This models "decides" to focus its model power on the big picture of the structure of the datasets: groups of points with alternating directions modeled by non-spherical covariance matrices. Those alternating directions roughly capture the alternating nature of the original sine signal.

The third model is also Bayesian Gaussian Mixture Model  with a Dirichlet process prior but this time the value of the concentration prior is higher giving the model more liberty to try to model the finer-grained structure of the data. The result is a mixture with a larger number of active components that is similar to the first model where we decided to fix the number of components to 10 arbitrarily.

Which models is best is a matter of subjective judgement:  do we want to favor models that only capture the big picture to summarize and explain most of the structure of the data while ignoring the details or do we prefer models that closely follow the high density regions of the signal?

The last two panels show how we can sample from the last two models. The resulting samples distributions do not look exactly like the original data distribution. The difference primarily stems from the approximation error we made by using a model that assumes that the data was generated by a finite number of Gaussian components instead of a noisy sine curve.